hadoop-hdds/docs/content/feature/Short-Circuit-Read.md

```diff
@@ -0,0 +1,78 @@
+  
+## Prerequisite
+
+Short-circuit local reads make use of a UNIX domain socket. This is a special path in the filesystem that allows the client and the DataNodes to communicate.
+
+The Hadoop native library `libhadoop.so` provides support to for Unix domain sockets. Please refer to Hadoop's [Native Libraries Guide](https://hadoop.apache.org/docs/stable/hadoop-project-dist/hadoop-common/NativeLibraries.html) for details.
+
+The Hadoop version used in Ozone is defined by `hadoop.version` in pom.xml. Before enabling short-circuit local reads, find the `libhadoop.so` from the corresponding version Hadoop release package, put it under one of the directories specified by Java `java.library.path` property. The default value of `java.library.path` depends on the OS and Java version. For example, on Linux with OpenJDK 8 it is `/usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib`.
+
+The `ozone checknative` command can be used to detect whether `libhadoop.so` can be found and loaded successfully by Ozone service.
+
+
+## Configuration
+
+Short-circuit local reads need to be configured on both the DataNode and the client. By default, it is disabled.
+
+```XML
+<property>
+   <name>ozone.client.read.short-circuit</name>
+   <value>false</value>
+   <description>Disable or enable the short-circuit local read feature.</description>
+</property>
+```
+
+It makes use of a UNIX domain socket, a special path in the filesystem. You will need to set a path to this socket. 
+
+```XML
+<property>
+   <name>ozone.domain.socket.path</name>
+   <value></value>
+   <description> This is a path to a UNIX domain socket that will be used for 
+      communication between the DataNode and local Ozone clients. 
+      If the string "_PORT" is present in this path, it will be replaced by the TCP port of the DataNode.
+   </description>
+</property>
+```
+
+The DataNode needs to be able to create this path. On the other hand, it should not be possible for any user except the Ozone user(user who launches Ozone service) or root to create this path. For this reason, paths under `/var/run` or `/var/lib` are often used.
+
+If you configure the `ozone.domain.socket.path` to some value, for example `/dir1/dir2/ozone_dn_socket`, please make sure that both `dir1` and `dir2` are existing directories, but the file `ozone_dn_socket` does not exist under `dir2`. `ozone_dn_socket` will be created by Ozone Datanode later during its startup.
+
+### Security Consideration
+
+To ensure data security and integrity, Ozone will follow the same rules as Hadoop to check permission on the `ozone.domain.socket.path` path as documented in [Socket Path Security](https://wiki.apache.org/hadoop/SocketPathSecurity).
+It will fail the `ozone.domain.socket.path` verification and disable the feature if the filesystem permissions of the specified path are inadequate.
+The verification failure message carries detail instruction about how to fix the problem. Following is an example, 
```

Comment:
These should be soft wrapped. The hard wrapping is not preserved in the rendered markdown and it makes the text harder to edit later. For easier editing set your markdown editor to use soft wraps.
```suggestion
To ensure data security and integrity, Ozone will follow the same rules as Hadoop to check permissions on the `ozone.domain.socket.path` path as documented in [Socket Path Security](https://wiki.apache.org/hadoop/SocketPathSecurity). It will fail the `ozone.domain.socket.path` verification and disable the feature if the filesystem permissions of the specified path are inadequate. The verification failure message carries detailed instructions about how to fix the problem. Following is an example: 
```